bundles/io/org.openhab.io.myopenhab/src/main/java/org/openhab/io/myopenhab/internal/MyOHClient.java

```diff
@@ -0,0 +1,559 @@
+	 * 
+	 */
+
+	public void setMyOHBaseUrl(String myOHBaseUrl) {
+		mMyOHBaseUrl = myOHBaseUrl;
+	}
+
+	/**
+	 * Set base local URL of openHAB
+	 * 
+	 * @param ohBaseUrl base local URL of openHAB in http://host:port form
+	 * 
+	 */
+
+	public void setOHBaseUrl(String ohBaseUrl) {
+		this.mOHBaseUrl = ohBaseUrl;
+	}
+
+	public String getOpenHABVersion() {
+		return mOpenHABVersion;
+	}
+
+	public void setOpenHABVersion(String mOpenHABVersion) {
+		this.mOpenHABVersion = mOpenHABVersion;
+	}
+	
+	public void setListener(MyOHClientListener mListener) {
+		this.mListener = mListener;
+	}
+
+	/*
+	 * An internal class which extends ContentExchange and forwards response
+	 * headers and data back to my.openHAB
+	 * 
+	 */
+	
+	private class MyOHExchange extends ContentExchange {
+		
+		private int mRequestId;
+		private HashMap<String, String> mResponseHeaders;
+
+		public MyOHExchange(int requestId) {
+			mRequestId = requestId;
+			mResponseHeaders = new HashMap<String, String>();
+		}
+
+		public void setRequestHeaders(JSONObject requestHeadersJson) {
+			Iterator<String> headersIterator = requestHeadersJson.keys();
+			// Convert JSONObject of headers into Header ArrayList
+			while (headersIterator.hasNext()) {
+				String headerName = headersIterator.next();
+				String headerValue;
+				try {
+					headerValue = requestHeadersJson.getString(headerName);
+					logger.debug("Jetty set header " + headerName + " = " + headerValue);
+					if (!headerName.equalsIgnoreCase("Content-Length"))
+						this.setRequestHeader(headerName, headerValue);
+				} catch (JSONException e) {
+					logger.error("Error processing request headers: {}", e.getMessage());
+				}
+			}			
+		}
+
+		public JSONObject getJSONHeaders() {
+			JSONObject headersJSON = new JSONObject();
+			try {
+				for (Map.Entry<String, String> responseHeader : mResponseHeaders.entrySet()) {
+						headersJSON.put(responseHeader.getKey(), responseHeader.getValue());
+				}
+			} catch (JSONException e) {
+				logger.error("Error forming response headers: {}", e.getMessage());
+			}
+			return headersJSON;
+		}
+
+		/*
+		 * This is old onResponseContent which used base64 encoding, keep it here commented for some time
+		 * The new onResponseContent uses 'responseContentBinary' to send response data
+		 
+		@Override
+		public void onResponseContent(Buffer content) {
+			logger.debug("Jetty received response content of size " + String.valueOf(content.length()));
+			JSONObject responseJson = new JSONObject();
+			String base64ResponseBody = Base64.encodeBytes(content.asArray());
+			try {
+				responseJson.put("id", mRequestId);
+				responseJson.put("body", base64ResponseBody);
+				if (this.getStatus() != STATUS_CANCELLING && this.getStatus() != STATUS_CANCELLED)
+					mSocket.emit("responseContent", responseJson);
+				logger.debug(String.format("Sent content to request %d", mRequestId));
+			} catch (JSONException e) {
+				e.printStackTrace();
+			}
+
+		}*/
```

Comment:
This code is left for reference until 1.7 version of bundle is deployed by all users.